### PR TITLE
Feature/fcla 98 consulting prototype #2

### PR DESCRIPTION
**Pull request by sbraggion**

### CODE REVIEW

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/IController.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import it.unibo.lss.fcla.consulting.application.presentation.IRequest
+
+/**
+ * @author Stefano Braggion
+ *
+ * Interface that each controller must implement
+ */
+interface IController {
+    fun execute(request: IRequest)
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/IController.kt:15]: No documentation for this fun?

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/FreelancerController.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                        request.availabilityDate,
+                        request.fromTime,
+                        request.toTime
+                    )
+                is GetFreelancerAvailabilityForDayRequest ->
+                    freelancerUseCases.getFreelancerAvailabilityForDay(
+                        request.freelancerId,
+                        request.availabilityDate
+                    )
+                else -> TODO()
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/FreelancerController.kt]: TODO?

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/ConsultingController.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import it.unibo.lss.fcla.consulting.application.presentation.consulting.ReceiveNutritionistConsultingRequest
+import it.unibo.lss.fcla.consulting.application.presentation.consulting.ReceivePhysiotherapyConsultingRequest
+import it.unibo.lss.fcla.consulting.domain.exceptions.ConsultingException
+import it.unibo.lss.fcla.consulting.usecases.ConsultingUseCases
+import it.unibo.lss.fcla.consulting.usecases.IPresenter
+
+/**
+ * @author Stefano Braggion
+ *
+ * This is a concrete implementation of [BaseController]. This class take the requests
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/ConsultingController.kt]: [BaseController]?

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/FreelancerController.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import it.unibo.lss.fcla.consulting.application.presentation.freelancer.GetFreelancerAvailabilityForDayRequest
+import it.unibo.lss.fcla.consulting.application.presentation.freelancer.UpdateFreelancerAvailabilityForDayRequest
+import it.unibo.lss.fcla.consulting.domain.exceptions.ConsultingException
+import it.unibo.lss.fcla.consulting.usecases.FreelancerUseCases
+import it.unibo.lss.fcla.consulting.usecases.IPresenter
+
+/**
+ * @author Stefano Braggion
+ *
+ * This is a concrete implementation of [BaseController]. This class take the requests
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/controllers/FreelancerController.kt]: [BaseController]?

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/persistence/ConsultingRepository.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package it.unibo.lss.fcla.consulting.application.persistence
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/persistence/ConsultingRepository.kt]: Is this the right package for a clean architecture? Maybe the repository should be put into a "adapter" package? See for example here https://github.com/carlphilipp/clean-architecture-example or here https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/presentation/IResponse.kt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package it.unibo.lss.fcla.consulting.application.presentation
+
+/**
+ * @author Stefano Braggion
+ *
+ * Interface that each response must implement
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/presentation/IResponse.kt:6]: "Interface that each response must implement", but "each response" of what? :)

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/persistence/EventStore.kt [hunk 7]
```diff
@@ -1,5 +1,6 @@
-package it.unibo.lss.fcla.consulting.common
+package it.unibo.lss.fcla.consulting.application.persistence
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/persistence/EventStore.kt]: Same question made for ConsultingRepository.

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/persistence/FreelancerRepository.kt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package it.unibo.lss.fcla.consulting.application.persistence
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/persistence/FreelancerRepository.kt]: Same question made for ConsultingRepository.

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/presentation/IRequest.kt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package it.unibo.lss.fcla.consulting.application.presentation
+
+/**
+ * @author Stefano Braggion
+ *
+ * Interface that each request must implement
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/presentation/IRequest.kt]: "Interface that each request must implement", but "each request" of what? :)

File: consulting/src/test/kotlin/it/unibo/lss/fcla/consulting/FreelancerControllerTest.kt [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package it.unibo.lss.fcla.consulting
+
+import io.kotest.core.spec.style.FreeSpec
+
+class FreelancerControllerTest : FreeSpec({
```
Comment (nlasagni) [inline consulting/src/test/kotlin/it/unibo/lss/fcla/consulting/FreelancerControllerTest.kt]: Maybe this test should be removed?

File: consulting/src/test/kotlin/it/unibo/lss/fcla/consulting/MockPresenter.kt [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+import it.unibo.lss.fcla.consulting.domain.exceptions.ConsultingException
+import it.unibo.lss.fcla.consulting.ui.IView
+import it.unibo.lss.fcla.consulting.usecases.IPresenter
+import it.unibo.lss.fcla.consulting.usecases.facades.BaseFacade
+
+/**
+ *
+ */
+class MockPresenter : IPresenter {
```
Comment (nlasagni) [inline consulting/src/test/kotlin/it/unibo/lss/fcla/consulting/MockPresenter.kt:11]: No documentation for this class?

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/usecases/IPresenter.kt [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+import it.unibo.lss.fcla.consulting.ui.IView
+import it.unibo.lss.fcla.consulting.usecases.facades.BaseFacade
+
+/**
+ * @author Stefano Braggion
+ *
+ * Interface that each presenter must implement
+ */
+interface IPresenter {
+    fun onResult(result: BaseFacade)
```
Comment (nlasagni) [inline consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/usecases/IPresenter.kt:17]: No documentation for these fun?

**Review by nlasagni — CHANGES_REQUESTED** [review]

**Review by Giordo94 — DISMISSED** [review]
Approved

File: consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/presentation/IRequest.kt (hunk 9 above)
Comment (sbraggion) [reply consulting/src/main/kotlin/it/unibo/lss/fcla/consulting/application/presentation/IRequest.kt]: each request of anything :)

**Review by nlasagni — APPROVED** [review]

**Pull request merged**